LocationUpdatesBackgroundKotlin/app/src/main/java/com/google/android/gms/location/sample/locationupdatesbackgroundkotlin/PermissionRequestFragment.kt

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (C) 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.gms.location.sample.locationupdatesbackgroundkotlin
+
+import android.Manifest
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.os.Bundle
+import android.provider.Settings
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.content.ContextCompat.checkSelfPermission
+import androidx.fragment.app.Fragment
+
+import com.google.android.gms.location.sample.locationupdatesbackgroundkotlin.databinding.FragmentPermissionRequestBinding
+import com.google.android.material.snackbar.Snackbar
+
+private const val TAG = "PermissionRequestFrag"
+
+/**
+ * Displays information about why a user should enable either the FINE location permission or the
+ * background location permission (depending on what is needed).
+ *
+ * Allows users to grant the permissions as well.
+ */
+class PermissionRequestFragment : Fragment() {
+
+    // Set by Activity for which type of permission to request (Fine or background).
+    private var permissionRequestType: PermissionRequestType? = null
+
+    private lateinit var binding: FragmentPermissionRequestBinding
+
+    private var activityListener: Callbacks? = null
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        arguments?.let {
```

Comment:
Maybe consider something like: `(arguments as? PermissionRequestType)?.let {` ?

Reply:
I need to cast the returned type from arguments.getSerializable(ARG_PERMISSION_REQUEST_TYPE) as PermissionRequestType, not the instance of arguments.  I guess I could do the code below but that seems more confusing to me than how it is written now. Any other ideas?

        (arguments?.getSerializable(ARG_PERMISSION_REQUEST_TYPE) as? PermissionRequestType)?.let {
            permissionRequestType = it
        }

Reply:
or maybe...

        permissionRequestType = arguments?.getSerializable(ARG_PERMISSION_REQUEST_TYPE) as PermissionRequestType

Reply:
x.x I was apparently too tired when I read through this. Yeah, the way you wrote it seems easiest to follow?
So yeah, let's just call it good.

Reply:
Ha ha :D 

Sounds good, changed.